C2LC-109: Use aria-disabled for the run button

chosww opened the pull request.

simonbates commented:
@chosww I've added a commit to `master` to make `npm audit` not fail the build. Could you merge `master` into this branch please?

https://github.com/codelearncreate/c2lc-coding-environment/commit/fef4c3939ab2619026d4ed2ade6c3ba52f8c0041

simonbates commented:
I've updated C2LC-109 to use a new `AriaDisablingButton` component for the run button. Could you please update this branch to use `AriaDisablingButton` for the Add and Delete editor buttons?

You will need to add some styling so that the buttons look disabled on hover and on active.

See: https://github.com/simonbates/c2lc-coding-environment/commit/88f342caa2554c4b1e404ebdcbc7f5ad36bb068a#diff-553ab764c97ca79a9fa98cd49952e754

chosww commented:
> I've updated C2LC-109 to use a new `AriaDisablingButton` component for the run button. Could you please update this branch to use `AriaDisablingButton` for the Add and Delete editor buttons?
> 
> You will need to add some styling so that the buttons look disabled on hover and on active.
> 
> See: [simonbates@88f342c#diff-553ab764c97ca79a9fa98cd49952e754](https://github.com/simonbates/c2lc-coding-environment/commit/88f342caa2554c4b1e404ebdcbc7f5ad36bb068a#diff-553ab764c97ca79a9fa98cd49952e754)



> I've updated C2LC-109 to use a new `AriaDisablingButton` component for the run button. Could you please update this branch to use `AriaDisablingButton` for the Add and Delete editor buttons?
> 
> You will need to add some styling so that the buttons look disabled on hover and on active.
> 
> See: [simonbates@88f342c#diff-553ab764c97ca79a9fa98cd49952e754](https://github.com/simonbates/c2lc-coding-environment/commit/88f342caa2554c4b1e404ebdcbc7f5ad36bb068a#diff-553ab764c97ca79a9fa98cd49952e754)

Sure, I will do that now 👍

chosww commented:
@simonbates should we move disabled styling to the global css file? Making disabled class for all the button components seems redundant.

simonbates commented:
Yes, I think having a general disabled button class with the rules in the global css file is a good idea. Thanks.

simonbates commented on src/AriaDisablingButton.js:
```diff
@@ -8,7 +8,6 @@ type AriaDisablingButtonProps = {
     onClick: (evt: any) => void,
     disabled: boolean,
     className?: string,
-    disabledClassName: string,
```
Please put this property back. We can still use a general CSS class in our global styles but provide it to the AriaDisabingButton instances as a property. This decouples the CSS class specifics from the component implementation.

chosww replied:
My bad, will do it now :)

simonbates commented on src/index.css:
```diff
@@ -42,6 +42,21 @@ button.command-block--pressed:hover {
+    fill: #777;
+}
+
+button.btn.button--disabled {
```
The #FFF background-color is specific to `btn-primary`. Change the selector to: `button.btn.button--disabled`.

simonbates commented on src/index.css:
```diff
@@ -42,6 +42,21 @@ button.command-block--pressed:hover {
+    cursor: default;
+}
+
+button.btn.button--disabled:active:focus {
```
The #FFF background-color is specific to `btn-primary`. Change the selector to: `button.btn-primary.button--disabled:active:focus`.
